src/common/arrstr.cpp

```diff
@@ -573,3 +579,169 @@ wxArrayString wxSplit(const wxString& str, const wxChar sep, const wxChar escape
+    class wxStringFragment
+    {
+    public:
+        wxStringFragment()
+            : type(wxFRAGMENT_TYPE_EMPTY)
+        {}
+
+        wxString             text;
+        double               value;
+        wxStringFragmentType type;
+    };
+
+
+    wxStringFragment GetFragment(wxString &text)
+    {
+        size_t digitStart = 0;
+        size_t digitLength = 0;
+        size_t alphaStart = 0;
+        size_t alphaLength = 0;
+        wxStringFragment fragment;
+
+        if (text.empty())
+            return fragment;
+
+        if (naturalNumeric.Matches(text))
+        {
+            naturalNumeric.GetMatch(&digitStart, &digitLength, 0);
+        }
+
+        if (naturalAlpha.Matches(text))
+        {
+            naturalAlpha.GetMatch(&alphaStart, &alphaLength, 0);
+        }
+
+
+        if (alphaStart == 0)
+        {
+            fragment.text = text.Mid(0, alphaLength);
+            fragment.value = 0;
+            fragment.type = wxFRAGMENT_TYPE_ALPHA;
+
+            text.erase(0, alphaLength);
+        }
+
+        if (digitStart == 0)
+        {
+            fragment.text = text.Mid(0, digitLength);
+            fragment.text.ToDouble(&fragment.value);
+            fragment.type = wxFRAGMENT_TYPE_DIGIT;
+
+            text.erase(0, digitLength);
+        }
+
+        return fragment;
+    }
+
+    int CompareFragmentNatural(const wxStringFragment &lhs, const wxStringFragment &rhs, bool noCase)
+    {
+        if ((lhs.type == wxFRAGMENT_TYPE_ALPHA) &&
+            (rhs.type == wxFRAGMENT_TYPE_ALPHA))
+        {
+            if (noCase)
+                return lhs.text.CmpNoCase(rhs.text);
+            else
+                return lhs.text.Cmp(rhs.text);
+        }
+
+        if ((lhs.type == wxFRAGMENT_TYPE_DIGIT) &&
+            (rhs.type == wxFRAGMENT_TYPE_DIGIT))
+        {
+            if (lhs.value == rhs.value)
+            {
+                return  0;
+            }
+
+            if (lhs.value < rhs.value)
+            {
+                return -1;
+            }
+
+            if (lhs.value > rhs.value)
+            {
+                return  1;
+            }
+        }
+
+        if ((lhs.type == wxFRAGMENT_TYPE_DIGIT) &&
+            (rhs.type == wxFRAGMENT_TYPE_ALPHA))
+        {
+            return -1;
+        }
+
+        if ((lhs.type == wxFRAGMENT_TYPE_ALPHA) &&
+            (rhs.type == wxFRAGMENT_TYPE_DIGIT))
+        {
+            return 1;
+        }
+
+        if (lhs.type == wxFRAGMENT_TYPE_EMPTY)
+        {
+            return -1;
+        }
+
+        if (rhs.type == wxFRAGMENT_TYPE_EMPTY)
+        {
+            return 1;
+        }
+
+        return 0;
+    }
+
+
+    inline int CompareNaturalFunction(const wxString &s1, const wxString &s2, bool noCase)
+    {
+        wxString lhs(s1);
+        wxString rhs(s2);
+        wxStringFragment fragL;
+        wxStringFragment fragR;
+
+        int comparison = 0;
+
+        while ((comparison == 0) && (lhs.Length() || rhs.Length()))
```

Comment:
`wxString::Length()` is a wxWidgets 1.x compatibility function, not to be used in the new code.

Please consider using `size()` instead.